pkg/dataobj/consumer/partition_processor.go

```diff
@@ -184,3 +202,78 @@ func (p *partitionProcessor) processRecord(record *kgo.Record) {
 		}
 	}
 }
+
+func (p *partitionProcessor) writeMetastores(backoff *backoff.Backoff, dataobjPath string) error {
```

Comment:
Should we have some kind of `dataobj/metastore` package that's responsible for building and operating on metastores?

Reply:
yes.